esp/logging/loggingagent/espserverloggingagent/loggingagent.hpp

```diff
@@ -42,6 +43,28 @@ enum ESPLogContentGroup
 
 static const char * const espLogContentGroupNames[] = { "ESPContext", "UserContext", "UserRequest", "UserResponse", "BackEndResponse", "", NULL };
 
+class CTransID : public CInterface, implements IInterface
+{
+    StringAttr seed;
+    unsigned __int64 seq;
+
+public:
+    IMPLEMENT_IINTERFACE;
+    CTransID(const char* _seed) : seed(_seed)
+    {
+        seq = 0;
+    };
+    virtual ~CTransID() {};
+
+    virtual void getTransID(StringArray& prefix, StringBuffer& id)
+    {//KW -> Rodrigo: not sure the following format is good enough.
```

Comment:
This implementation is very practical, but the order in which the elements are inserted dictate the trxid, I think the logagent should dictate the format, and it should be consistent. 

The transactional ESP should provide a DATE-TIME stamp, a unique ESP instance identifier, and a method name. 
let's change this function to accept a key/value pair container in which the esp provide those elements, and can provide more in the future without having to change the signature. 
a Trxid consisting of <Date Time>-<Method>-<Unique ESP ID>-Sequence(seed+seq) should be serviceable.

A thing  to consider:
There's a chance the logagent was not able to fetch a seed from the back-end, we need to be able to distinguish trxids seeded by back-end, and locally seeded ids. It could be as simple as adding an 'X' to the sequence.